### DISCUSSION TITLE
Why I can't find cpufreq on VM?

### DISCUSSION THREAD

**Discussion by peng6662001** [cloud-hypervisor/cloud-hypervisor#6254] [General]
I can find /sys/devices/system/cpu/cpu0/cpufreq on the host machine, but not on a VM with ubuntu 22.04.
I compiled the Linux kernel with the 6.6.0-rc6 configuration file from the host on VM and installed cpufrequtils.

**Comment by liuw** [comment]
If that capability is not available to the VM, the driver inside the VM will not be loaded and tools won't be of any use.

In general, a guest VM shouldn't be able to control the cpu frequency of the whole system -- other VMs may want to run at full speed, so only the host knows the whole picture and only the host should have control over the cpu power state.

And it doesn't make sense for the VM to throttle itself either.
